modules/space/rendering/renderablehabitablezone.cpp

```diff
@@ -0,0 +1,278 @@
+        );
+    }
+    _teff.onChange([this]() { computeZone(); });
+    addProperty(_teff);
+
+    if (dictionary.hasKey(LuminosityInfo.identifier)) {
+        _luminosity = static_cast<float>(
+            dictionary.value<double>(LuminosityInfo.identifier)
+        );
+    }
+    _luminosity.onChange([this]() { computeZone(); });
+    addProperty(_luminosity);
+
+    if (dictionary.hasKey(OptimisticInfo.identifier)) {
+        _showOptimistic = dictionary.value<bool>(OptimisticInfo.identifier);
+    }
+    addProperty(_showOptimistic);
+
+    // Make parent's size related properties read only. We want to set them based on the
+    // given temperature and luminosity
+    _size.setReadOnly(true);
+    _width.setReadOnly(true);
+
+    computeZone();
+}
+
+void RenderableHabitableZone::render(const RenderData& data, RendererTasks&) {
+    _shader->activate();
+
+    glm::dmat4 modelTransform =
+        glm::translate(glm::dmat4(1.0), data.modelTransform.translation) *
+        glm::dmat4(data.modelTransform.rotation) *
+        glm::scale(glm::dmat4(1.0), glm::dvec3(data.modelTransform.scale));
+
+    glm::dmat4 modelViewTransform = data.camera.combinedViewMatrix() * modelTransform;
+
+    _shader->setUniform(
+        _uniformCache.modelViewProjection,
+        data.camera.projectionMatrix() * glm::mat4(modelViewTransform)
+    );
+    _shader->setUniform(_uniformCache.width, _width);
+    _shader->setUniform(_uniformCache.opacity, _opacity);
+    _shader->setUniform(_uniformCache.conservativeBounds, _conservativeBounds);
+    _shader->setUniform(_uniformCache.showOptimistic, _showOptimistic);
+
+    ghoul::opengl::TextureUnit unit;
+    unit.activate();
+    _texture->bind();
+    _shader->setUniform(_uniformCache.texture, unit);
+
+    glEnablei(GL_BLEND, 0);
+    glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
+    glDepthMask(false);
+    glDisable(GL_CULL_FACE);
+
+    _plane->render();
+
+    _shader->deactivate();
+
+    // Restores GL State
+    global::renderEngine->openglStateCache().resetBlendState();
+    global::renderEngine->openglStateCache().resetDepthState();
+    global::renderEngine->openglStateCache().resetPolygonAndClippingState();
+}
+
+void RenderableHabitableZone::initializeShader() {
+    _shader = global::renderEngine->buildRenderProgram(
+        "HabitableZoneProgram",
+        absPath("${MODULE_SPACE}/shaders/habitablezone_vs.glsl"),
+        absPath("${MODULE_SPACE}/shaders/habitablezone_fs.glsl")
+    );
+    ghoul::opengl::updateUniformLocations(*_shader, _uniformCache, UniformNames);
+}
+
+void RenderableHabitableZone::updateUniformLocations() {
+    ghoul::opengl::updateUniformLocations(*_shader, _uniformCache, UniformNames);
+}
+
+void RenderableHabitableZone::computeZone() {
+    glm::vec4 distancesInAu = computeKopparapuZoneBoundaries(_teff, _luminosity);
+    constexpr float AU = static_cast<float>(distanceconstants::AstronomicalUnit);
```

Comment:
`constexpr const float`?

Reply:
Also;  I'd check if the `float` is actually sufficient.  It only has 7 or 8 significant digits and these values might run out of precision.  Maybe better to do the calculations in `double` and then cast to `float` right at the end